components/director/internal/tenantmapping/handler_test.go

```diff
@@ -906,3 +1517,43 @@ func getMockContextProvider() *automock.ObjectContextProvider {
 	provider := &automock.ObjectContextProvider{}
 	return provider
 }
+
+func assertExtra(t *testing.T, expectedBody, actualBody string) {
+	expectedExtra := gjson.Get(expectedBody, "extra").Map()
+	actualExtra := gjson.Get(strings.TrimSpace(actualBody), "extra").Map()
+	for k, v := range expectedExtra {
+		if k != "scope" {
+			require.Equal(t, v.Str, actualExtra[k].Str)
+		}
+	}
+
+	expectedScopes := strings.Split(expectedExtra["scope"].String(), " ")
+	actualScopes := strings.Split(actualExtra["scope"].String(), " ")
+	require.True(t, scopesAreEqual(expectedScopes, actualScopes))
+}
+
+func scopesAreEqual(expectedScopes, actualScopes []string) bool {
```

Comment:
Why not use require.Equal?

Reply:
As the order in which the scopes will be provided is not deterministic we may receive different results on separate executions of the tests. If we set expectedScopes to be []string{"test1", "test2", "test3"} and the actual scopes are []string{"test1", "test3", "test2"} the result from require.Equal will fail as the order of elements is not the same.

Reply:
Okay, makes sense, then you can use `require.ElementsMatch`